### PR TITLE
NLU-3780 Catalan minimal entity support

### PR DESCRIPTION
**Pull request by colm-dillon**

### CODE REVIEW

File: Python/libraries/recognizers-date-time/recognizers_date_time/date_time/catalan/common_configs.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+from .time_extractor_config import CatalanTimeExtractorConfiguration
+from .time_parser_config import CatalanTimeParserConfiguration
+
+
+class CatalanBaseDateParser(BaseDateParser):
+
+    def __init__(self, config: DateParserConfiguration):
+        super().__init__(config)
+
+    def extract(self, source: str, reference: datetime = None) -> List[ExtractResult]:
```
Comment (colm-dillon) [inline Python/libraries/recognizers-date-time/recognizers_date_time/date_time/catalan/common_configs.py:29]: My solution to not calling the duration extraction.

File: Python/libraries/recognizers-number/recognizers_number/number/catalan/parsers.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+        # The following piece of code is needed to compute the fraction pattern number+'y medio'
+        # e.g. 'cinco y medio' ('five and a half') where the numerator is omitted in Catalan.
+        # It works by inserting the numerator 'un' ('a') in the list result
+        # so that the pattern is correctly processed.
+        # if len(result) > 2:
+        #     if result[len(result) - 1] == CatalanNumeric.OneHalfTokens[1] and \
+        #             result[len(result) - 2] == CatalanNumeric.WordSeparatorToken:
+        #         result[len(result) - 2] = CatalanNumeric.WrittenFractionSeparatorTexts[0]
+        #         result.insert(len(result) - 1, CatalanNumeric.OneHalfTokens[0])
```
Comment (Conor-Keaney) [inline Python/libraries/recognizers-number/recognizers_number/number/catalan/parsers.py]: Is this supposed to be uncommented?

File: Python/libraries/recognizers-number/recognizers_number/number/catalan/parsers.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+        ordinal_number_map: Dict[str, int] = dict(
+            CatalanNumeric.OrdinalNumberMap)
+        # for prefix_key in CatalanNumeric.PrefixCardinalMap:
+        #     for suffix_key in CatalanNumeric.SuffixOrdinalMap:
+        #         if not prefix_key+suffix_key in ordinal_number_map:
+        #             prefix_value = CatalanNumeric.PrefixCardinalMap[prefix_key]
+        #             suffix_value = CatalanNumeric.SuffixOrdinalMap[suffix_key]
+        #             ordinal_number_map[prefix_key +
+        #                                suffix_key] = prefix_value*suffix_value
```
Comment (Conor-Keaney) [inline Python/libraries/recognizers-number/recognizers_number/number/catalan/parsers.py]: Maybe removing commented code, or providing a comment / TODO above it

File: Python/libraries/recognizers-date-time/recognizers_date_time/date_time/catalan/merged_parser_config.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        CatalanCommonDateTimeParserConfiguration.__init__(self)
+        self._time_zone_parser = config.time_zone_parser
+        self._equal_regex = RegExpUtility.get_safe_reg_exp(BaseDateTime.EqualRegex)
+        self._suffix_after = RegExpUtility.get_safe_reg_exp(f'^[.]')
+        self._year_regex = RegExpUtility.get_safe_reg_exp(
+            CatalanDateTime.YearRegex)
+        self._around_regex = RegExpUtility.get_safe_reg_exp(f'^[.]')
+        self._before_regex = RegExpUtility.get_safe_reg_exp(f'^[.]')
+        self._after_regex = RegExpUtility.get_safe_reg_exp(f'^[.]')
+        self._since_regex = RegExpUtility.get_safe_reg_exp(f'^[.]')
```
Comment (Conor-Keaney) [inline Python/libraries/recognizers-date-time/recognizers_date_time/date_time/catalan/merged_parser_config.py:63]: Are these unused?

File: Python/libraries/recognizers-date-time/recognizers_date_time/date_time/catalan/date_parser_config.py [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+            CatalanDateTime.WeekDayRegex)
+        self._last_regex = RegExpUtility.get_safe_reg_exp(f'^[.]')
+        self._this_regex = RegExpUtility.get_safe_reg_exp(f'^[.]')
+        self._week_day_of_month_regex = RegExpUtility.get_safe_reg_exp(f'^[.]')
+        self._for_the_regex = RegExpUtility.get_safe_reg_exp(f'^[.]')
+        self._week_day_and_day_of_month_regex = RegExpUtility.get_safe_reg_exp(
+            CatalanDateTime.WeekDayAndDayOfMonthRegex)
+        self._week_day_and_day_regex = RegExpUtility.get_safe_reg_exp(
+            CatalanDateTime.WeekDayAndDayRegex)
+        self._relative_month_regex = RegExpUtility.get_safe_reg_exp(f'^[.]')
```
Comment (Conor-Keaney) [inline Python/libraries/recognizers-date-time/recognizers_date_time/date_time/catalan/date_parser_config.py:176]: Is this temporary or might there be another way to mark regexes that are unused without it failing on a defined None?

Comment (colm-dillon) [reply Python/libraries/recognizers-date-time/recognizers_date_time/date_time/catalan/date_parser_config.py:176]: To be fair I've taken these directly from @samhickey25's spike. I don't mind this approach, seems to be done in other languages in places like Portuguese, German etc.

File: Python/libraries/recognizers-date-time/recognizers_date_time/date_time/catalan/merged_parser_config.py (hunk 4 above)
Comment (colm-dillon) [reply Python/libraries/recognizers-date-time/recognizers_date_time/date_time/catalan/merged_parser_config.py:63]: See above

File: Python/libraries/recognizers-number/recognizers_number/number/catalan/parsers.py (hunk 2 above)
Comment (colm-dillon) [reply Python/libraries/recognizers-number/recognizers_number/number/catalan/parsers.py]: @samhickey25 did you remember this snippet? I might leave it there with a TODO if we want to keep it.

Comment (samhickey25) [reply Python/libraries/recognizers-number/recognizers_number/number/catalan/parsers.py]: the commented code can be removed

**Review by Conor-Keaney — APPROVED** [review]
LGTM

**Review by samhickey25 — APPROVED** [review]

File: Specs/Sequence/Catalan/PhoneNumberModel.json
Comment (samhickey25) [inline Specs/Sequence/Catalan/PhoneNumberModel.json:1]: what's this?

File: Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_date.py [hunk 6]
```diff
@@ -235,6 +235,8 @@ def extract(self, source: str, reference: datetime = None) -> List[ExtractResult
         tokens.extend(self.basic_regex_match(source))
         tokens.extend(self.implicit_date(source))
         tokens.extend(self.number_with_month(source, reference))
+
+        # TODO: Disable for Catalan
         tokens.extend(self.relative_duration_date(source, reference))
```
Comment (andrew-gradinari) [inline Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_date.py:240]: Should this still be disabled?

**Review by andrew-gradinari — APPROVED** [review]

**Review by KanchanKumar12 — APPROVED** [review]
LGTM

File: Python/libraries/recognizers-date-time/recognizers_date_time/date_time/catalan/date_extractor_config.py [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+            RegExpUtility.get_safe_reg_exp(date_extractor_8),
+            RegExpUtility.get_safe_reg_exp(date_extractor_9),
+            RegExpUtility.get_safe_reg_exp(CatalanDateTime.DateExtractor10),
+        ]
+
+        self._implicit_date_list = [
+            RegExpUtility.get_safe_reg_exp(CatalanDateTime.OnRegex),
+            RegExpUtility.get_safe_reg_exp(CatalanDateTime.RelaxedOnRegex),
+            RegExpUtility.get_safe_reg_exp(CatalanDateTime.SpecialDayRegex),
+            # RegExpUtility.get_safe_reg_exp(CatalanDateTime.ThisRegex),
```
Comment (kevinwalshgen) [inline Python/libraries/recognizers-date-time/recognizers_date_time/date_time/catalan/date_extractor_config.py:166]: Do you want to leave a comment here as to why these are left commented out?

File: Specs/NumberWithUnit/Catalan/CurrencyModel.json [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+          "unit": "Lliura"
+        },
+        "Start": 87,
+        "End": 91
+      }
+    ]
+  },
+   {
+    "Input": "10 $ no són gaires, ni 10 €",
+    "NotSupported": "javascript, python, java",
```
Comment (kevinwalshgen) [inline Specs/NumberWithUnit/Catalan/CurrencyModel.json]: So this only supports the above case and none of the others?

File: Python/libraries/recognizers-number/recognizers_number/number/catalan/parsers.py [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+
+from recognizers_text.utilities import RegExpUtility
+from recognizers_text.culture import Culture
+from recognizers_text.parser import ParseResult
+from recognizers_number.culture import CultureInfo
+from recognizers_number.number.parsers import NumberParserConfiguration
+from recognizers_number.resources.catalan_numeric import CatalanNumeric
+
+
+class CatalanNumberParserConfiguration(NumberParserConfiguration):
```
Comment (kevinwalshgen) [inline Python/libraries/recognizers-number/recognizers_number/number/catalan/parsers.py]: If you use the BaseNumberParserConfiguration you wouldn't need to add `normalize_token_set`, `resolve_composite_number` and `get_lang_specific_int_value`

File: Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_date.py (hunk 6 above)
Comment (colm-dillon) [reply Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_date.py:240]: I have disabled it, I've overwritten the extract method [here](https://github.com/purecloudlabs/Recognizers-Text/blob/81e2ba6ffc054d3efc6bd10d5285a22d556c69d2/Python/libraries/recognizers-date-time/recognizers_date_time/date_time/catalan/common_configs.py#L30)

File: Specs/NumberWithUnit/Catalan/CurrencyModel.json (hunk 8 above)
Comment (colm-dillon) [reply Specs/NumberWithUnit/Catalan/CurrencyModel.json]: Ahh I was surprised my tests passed, good spot. Let's give them another run.

File: Specs/Sequence/Catalan/PhoneNumberModel.json
Comment (colm-dillon) [reply Specs/Sequence/Catalan/PhoneNumberModel.json:1]: Deleting.

File: Python/libraries/datatypes-timex-expression/setup.py [hunk 10]
```diff
@@ -11,7 +11,7 @@ def read(fname):
 
 
 NAME = 'datatypes_timex_expression_genesys'
-VERSION = '1.0.66'
+VERSION = '1.0.67'
```
Comment (samhickey25) [inline Python/libraries/datatypes-timex-expression/setup.py]: should probably be bumped to 1.0.70

Comment (colm-dillon) [reply Python/libraries/datatypes-timex-expression/setup.py]: Yes my mistake. Rebuilding with .70 now

**Pull request merged**